### ISSUE TITLE
Enum as int

### ISSUE THREAD

**Issue by fenomeno83** [ChilliCream/hotchocolate#2415]
Hi.
I use hot chocolate pure code first approach, so I don't need to write graphql schema and types.

For example my query returns this model:

    public class ReturnedModel
    {
        public string SomeInfo {get;set;}
        public TestEnum TestEnum {get;set;}
    }

    public enum TestEnum
    {
      FirstEnum=100,
      SecondEnum=200
    }

Comparing to a classic api, when the returned model has an enum property, the api returns integer value (for example 100), but hot chocolate returns string (for example "FirstEnum")

How can I return integer value?

Thanks

**Comment by michaelstaib** [comment]
With GraphQL you should return a proper enum and not an int. The enum is automatically inferred.

**Comment by fenomeno83** [comment]
Thanks for you reply.
But consider that I have an enum with groups ids that I can save on database.
Is there no way to convert enum on its integer and return it?

**Comment by michaelstaib** [comment]
Sure, there are ways to do it but it is considered bad practice. Why would you save on the database? If you really want to optimize bandwidth than consider using persisted queries in which case you only transfer a hash instead of the query.

Also, I would strongly suggest reading into GraphQL concepts and not compare them to REST.

Here are some approaches you can use (although I recommend against it):

```csharp
public class ReturnedModel
{
    public string SomeInfo {get;set;}
    [GraphQLType(typeof(IntType))]
    public TestEnum TestEnum {get;set;}
}
```

Define that the type is an int type on the schema basis:

```csharp
...BindRuntimeType<TestEnum, IntType>()...
```

If you just want to change the serialization result you can reimplement the result serializer... but in this case, you also need to handed deserialization for input types.

**Comment by fenomeno83** [comment]
Thanks for your reply.
My c# model is taken from external nuget package library, so I can't add GraphQLType data annotation in properties.

I've tried to add in startup
.BindClrType<TestEnum, IntType>()
but I have this error
 "message": "The internal resolver value could not be converted to a valid value of `Int`.",
but schema type report "int" correctly

But i'd like to generalize it for all enums, so I tried this
.BindClrType<Enum, IntType>()
it doesn't generated error but don't do nothing in my output

Can I solve?
